core/src/main/java/io/aiven/klaw/service/KafkaConnectControllerService.java

```diff
@@ -464,6 +465,18 @@ public ApiResponse deleteConnectorRequests(String topicId) throws KlawException
     }
   }
 
+  public ApiResponse restartConnector(KafkaConnectorRestartModel kafkaConnectorRestartModel) {
+    int tenantId = commonUtilsService.getTenantId(getUserName());
+    try {
+      return clusterApiService.restartConnector(kafkaConnectorRestartModel, tenantId);
```

Comment:
This is where I should have asked for the permission check, and I cna see that you have already created the right permissions etc so I think here would be a good place to check for it.

Reply:
updated. good one.